src/RelayServer.ts

```diff
@@ -377,40 +353,50 @@ export class RelayServer extends EventEmitter {
                 )
             );
         }
-        log.debug(
-            'RequestFees - allowForSponsoredTx ',
-            this.config.allowForSponsoredTx
-        );
-        if (!this.config.allowForSponsoredTx) {
+
+        const { workerFeePercentage } = this.config;
+        log.debug(`RelayServer - workerFeePercentage: ${workerFeePercentage}`);
+
+        const sponsoredTxFee: BigNumber = normaliseFraction({ fraction: workerFeePercentage }).absoluteValue();
+        if (sponsoredTxFee.isGreaterThan(0)) {
             // we need to convert tokenAmount back into RBTC and compare its value with maxPossibleGas
             // if the value is lower than maxPossibleGas, we should throw an error
-            // TODO: we may need add some percentage fee at some point.
             const tokenAmountInGas = getGas(
-                getRBTCWeiFromRifWei(
-                    toBN(req.relayRequest.request.tokenAmount)
-                ),
+                getRBTCWeiFromRifWei(toBN(req.relayRequest.request.tokenAmount)),
                 toBN(req.relayRequest.relayData.gasPrice)
             );
-            const isTokenAmountAcceptable =
-                tokenAmountInGas.gte(maxPossibleGas);
+
+            const gasAsBigNumber: BigNumber = new BigNumber(maxPossibleGas.toString());
+            const feeValueInWei: BigNumber = gasAsBigNumber.multipliedBy(sponsoredTxFee);
+            const feeValue: BN = fractionToBN({
+                fraction: feeValueInWei,
+                precision: -RBTC_CHAIN_DECIMALS
+            });
+            const gasWithFees: BN = maxPossibleGas.add(feeValue);
+            const isTokenAmountAcceptable = tokenAmountInGas.gte(gasWithFees);
+
             log.debug(
                 'RequestFees - isTokenAmountAcceptable? ',
                 isTokenAmountAcceptable
             );
+
             if (!isTokenAmountAcceptable) {
                 log.warn(
                     'TokenAmount in gas agreed by the user',
                     tokenAmountInGas.toString()
                 );
                 log.warn(
-                    'MaxPossibleGas required by the transaction',
-                    maxPossibleGas.toString()
+                    'MaxPossibleGas including fees required by the transaction',
+                    gasWithFees.toString()
                 );
                 throw new Error(
-                    'User agreed to spend lower than what the transaction may require.'
+                    'User agreed to spend lower than what the transaction may require.' // @note should we extract strings to some constants file to make em more reusable? (eg for tests)
```

Comment:
Yes, we could start with the ones you're currently using.

Reply:
done